craft_store/auth.py

```diff
@@ -0,0 +1,87 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store Authentication Store."""
+
+import base64
+import logging
+
+import keyring
+import keyring.errors
+
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+class Auth:
+    """Auth wraps around the keyring to store credentials.
+
+    The application_name and host are used as key/values in the keyring to set,
+    get and delete credentials.
+
+    Credentials are base64 encoded into the keyring and decoded on retrieval.
+
+    :ivar application_name: name of the application using this library.
+    :ivar host: specific host for the store used.
+    """
+
+    def __init__(self, application_name: str, host: str) -> None:
+        """Initialize Auth.
+
+        :param application_name: name of the application using this library.
+        :param host: specific host for the store used.
+        """
+        self.application_name = application_name
+        self.host = host
+
+    def set_auth(self, auth: str) -> None:
+        """Store credentials in the keyring.
+
+        :param auth: token to store.
```

Comment:
I wonder if the parameter should be called `token` instead.

Reply:
+1. Or `credentials`.